docs/DeveloperGuide.adoc

```diff
@@ -25,242 +23,382 @@ Refer to the guide <<SettingUp#, here>>.
-* does not depend on any of the other three components.
-
-[NOTE]
-As a more OOP model, we can store a `Tag` list in `Address Book`, which `Person` can reference. This would allow `Address Book` to only require one `Tag` object per unique `Tag`, instead of each `Person` needing their own `Tag` object. An example of how such a model may look like is given below. +
- +
-image:BetterModelClassDiagram.png[]
+The `Model` component is independent of the other three primary components of NUStudy.
+It stores application data – notes, questions and revision tasks – as well as a `UserPrefs` object
+representing the user's preferences. It also exposes an unmodifiable `ObservableList<Note>` that is bound to the UI,
+so any data changes are immediately user-visible.
 
 [[Design-Storage]]
-=== Storage component
+=== `Storage`
 
-.Structure of the Storage Component
+.Class diagram of the `Storage` component
 image::StorageClassDiagram.png[]
 
-*API* : link:{repoURL}/src/main/java/seedu/address/storage/Storage.java[`Storage.java`]
+_API:_ link:{repoURL}/src/main/java/seedu/address/storage/Storage.java[`Storage.java`]
 
-The `Storage` component,
+The `Storage` component reads and writes `UserPrefs` objects and NUStudy data in JSON format.
 
-* can save `UserPref` objects in json format and read it back.
-* can save the Address Book data in json format and read it back.
+== Implementation
 
-[[Design-Commons]]
-=== Common classes
+This section describes some finer details on how certain features are implemented.
 
-Classes used by multiple components are in the `seedu.addressbook.commons` package.
+// tag::lecimage[]
+=== Image support for lecture notes
+==== Implementation
 
-== Implementation
+Instead of a custom class to represent an image, JavaFX's `scene.image.Image` is used instead.
+This is out of necessity, since an `ImageView` is needed to display an image in the GUI and it requires
+an `Image` object, not just a `String` path to the image. No significant coupling is introduced
+by this choice because `Image` objects can exist without a GUI to display them.
 
-This section describes some noteworthy details on how certain features are implemented.
+The dialog brought up to let the user choose a file returns `null` if the dialog is closed
+_without_ choosing a file (i.e. clicking the close button). Treating `null` as "no image" would surprise the user who
+decides to edit a lecture note's image but then decides not to – the image would simply disappear without
+warning. Therefore the value `i/none` has to be explicitly provided to remove the image, and `EditNoteDescriptor`
+has an additional field `isImageRemoved` to track this. If the image comes out as `null` but `i/none` was
+not provided, the image is unchanged.
 
-// tag::undoredo[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+To avoid a dependency of the `Logic` module on the `UI` module when selecting an image, there is the interface
```

Comment:
would it be better to orientate the reader on how this feature works first before jumping into handling the exception case?

Reply:
That already happened in the previous paragraph.